openshift/topics/tutorials.adoc

```diff
@@ -842,6 +797,541 @@ To provide _testuser_ `view` privileges for the _sso-app-demo_, use the OpenShif
+. Build both the link:https://github.com/keycloak/keycloak-quickstarts/tree/latest/service-jee-jaxrs[service-jee-jaxrs] and link:https://github.com/keycloak/keycloak-quickstarts/tree/latest/app-jee-jsp[app-jee-jsp] applications.
+.. Build the `service-jee-jaxrs` application.
++
+[source,bash,subs="attributes+,macros+"]
+----
+$ cd keycloak-quickstarts/service-jee-jaxrs/
+----
++
+[source,bash,subs="attributes+,macros+"]
+----
+$ mvn clean package -DskipTests
+[INFO] Scanning for projects...
+[INFO]
+[INFO] ------------------------------------------------------------------------
+[INFO] Building Keycloak Quickstart: service-jee-jaxrs 3.1.0.Final
+[INFO] ------------------------------------------------------------------------
+...
+[INFO] ------------------------------------------------------------------------
+[INFO] BUILD SUCCESS
+[INFO] ------------------------------------------------------------------------
+[INFO] Total time: 2.153 s
+[INFO] Finished at: 2017-06-26T12:06:12+02:00
+[INFO] Final Memory: 25M/241M
+[INFO] ------------------------------------------------------------------------
+----
+.. *Comment out* the `app-jee-jsp/config/keycloak.json` requirement of the `maven-enforcer-plugin` plugin and build the `app-jee-jsp` application.
++
+[source,bash,subs="attributes+,macros+"]
+----
+service-jee-jaxrs]$ cd ../app-jee-jsp/
+----
++
+[source,bash,subs="attributes+,macros+"]
+----
+app-jee-jsp]$ sed -i /\<executions\>/s/^/\<\!--/ pom.xml
+----
++
+[source,bash,subs="attributes+,macros+"]
+----
+app-jee-jsp]$ sed -i '/\(<\/executions>\)/a\-->' pom.xml
+----
++
+[source,bash,subs="attributes+,macros+"]
+----
+app-jee-jsp]$ mvn clean package -DskipTests
+[INFO] Scanning for projects...
+[INFO]
+[INFO] ------------------------------------------------------------------------
+[INFO] Building Keycloak Quickstart: app-jee-jsp 3.1.0.Final
+[INFO] ------------------------------------------------------------------------
+...
+[INFO] Building war: /tmp/github/keycloak-quickstarts/app-jee-jsp/target/app-jsp.war
+[INFO] ------------------------------------------------------------------------
+[INFO] BUILD SUCCESS
+[INFO] ------------------------------------------------------------------------
+[INFO] Total time: 3.018 s
+[INFO] Finished at: 2017-06-26T12:22:25+02:00
+[INFO] Final Memory: 35M/310M
+[INFO] ------------------------------------------------------------------------
+----
++
+[IMPORTANT]
+====
+The link:https://github.com/keycloak/keycloak-quickstarts/tree/latest/app-jee-jsp[app-jee-jsp] quickstart requires to configure the adapter, and adapter configuration file (`keycloak.json`) to be present at the `config/` directory in the root of the quickstart to successfully build the quickstart. But since this example configures the adapter later via selected environment variables available for the EAP 6.4 / 7.1 for OpenShift image, it is not necessary to specify the form of `keycloak.json` adapter configuration file at this moment.
```

Comment:
"...quickstart requires you to configure..." "...and that the adapter configuration file...is present in the..."